src/lib.rs

```diff
@@ -123,27 +127,30 @@ mod client;
 pub mod configuration;
 pub mod error;
 mod macros;
+pub mod tracker;
 
 #[cfg(test)]
 #[path = "tests/mod.rs"]
 mod tests;
 
 /// Types that can provide an iterator of tags for a metric.
 ///
-/// This is automatically implemented for anything that implements `IntoIterator<Item = S>` where `S: AsRef<str>`.
+/// This is automatically implemented for anything that implements `AsRef<[S]>` where `S: AsRef<str>`.
 ///
 /// In other words, you can pass a `&[&str]` (best), `Vec<&str>`, `&[String]`, `Vec<String>`, `&[Cow<str>]`, `Vec<Cow<str>>`, etc.
 ///
 /// **If you'd like to pass in empty tags, use the [EMPTY_TAGS] constant.**
-pub trait TagsProvider: IntoIterator<Item = Self::Tag> {
-    type Tag: AsRef<str>;
+pub trait TagsProvider<S>
+where
+    Self: AsRef<[S]>,
+    S: AsRef<str>,
+{
 }
-impl<I, S> TagsProvider for I
+impl<S, T> TagsProvider<S> for T
```

Comment:
Why do we need this change to the type parameter? I don't like much the generic `S` polluting all the public APIs...

Reply:
Agreed, but unfortunately this is the only way to accept different string types (`&str`, `&'static str`, `String`, etc)

Reply:
I think those generic bounds could do with a more descriptive name though

Reply:
> Unfortunately this is the only way to accept different string types (&str, &'static str, String, etc)

But didn't we do that already? 🤔

Reply:
Yes we did do that already in the public-facing interface, however this was not implemented for this PR yet (specifically in the tags cardinality tracker), which is what I've implemented in this change :D